### PR TITLE
[RFR] Add color prop to AppBar

### LINKED ISSUE #4396: AppBar should not use secondary color
Currently `AppBar` uses secondary color as backgroud: 

`<MuiAppBar className={className} color="secondary" {...rest}>`

Material Design suggests:

> A secondary color provides more ways to accent and distinguish your product. Having a secondary color is optional, and should be applied sparingly to accent select parts of your UI.
> 
> Secondary colors are best for:
> 
> Floating action buttons
> Selection controls, like sliders and switches
> Highlighting selected text
> Progress bars
> Links and headlines

So I think `AppBar` should use primary color by default.

### PR DESCRIPTION
**Pull request by WiXSL**
Fixes #4396

### CODE REVIEW

File: packages/ra-ui-materialui/src/layout/AppBar.js [hunk 1]
```diff
@@ -47,11 +47,40 @@ const useStyles = makeStyles(
...
 );
-
+/**
+ * The AppBar component renders a custom MuiAppBar.
+ *
+ * @param {Function} children Must be a function which will be called with the mutate callback
+ * @param {Object} classes CSS class names
+ * @param {string} className CSS class applied to the MuiAppBar component
+ * @param {string} color The color of the AppBar
+ * @param {Object} logo //
```
Comment (djhi) [inline packages/ra-ui-materialui/src/layout/AppBar.js]: Would you mind removing this very old prop which is not used at all anymore ?

File: packages/ra-ui-materialui/src/layout/AppBar.js [hunk 2]
```diff
@@ -47,11 +47,40 @@ const useStyles = makeStyles(
...
+ *        </AppBar>
+ *    );
+ *};
+ */
 const AppBar = ({
     children,
     classes: classesOverride,
     className,
+    color = 'secondary',
     logo,
```
Comment (djhi) [inline packages/ra-ui-materialui/src/layout/AppBar.js]: Please remove it too

Comment (WiXSL) [reply packages/ra-ui-materialui/src/layout/AppBar.js]: Done!

File: packages/ra-ui-materialui/src/layout/AppBar.js [hunk 3]
```diff
@@ -47,12 +47,39 @@ const useStyles = makeStyles(
     }),
     { name: 'RaAppBar' }
 );
-
+/**
+ * The AppBar component renders a custom MuiAppBar.
+ *
+ * @param {Function} children Must be a function which will be called with the mutate callback
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/AppBar.js]: No, this is wrong. Children should be a React node.

File: packages/ra-ui-materialui/src/layout/AppBar.js [hunk 4]
```diff
@@ -47,12 +47,39 @@ const useStyles = makeStyles(
...
+ * @param {Object} classes CSS class names
+ * @param {string} className CSS class applied to the MuiAppBar component
+ * @param {string} color The color of the AppBar
+ * @param {Object} logout The logout button component that will be pass to the UserMenu component
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/AppBar.js]: Not Object, but component

File: packages/ra-ui-materialui/src/layout/AppBar.js [hunk 5]
```diff
@@ -47,12 +47,39 @@ const useStyles = makeStyles(
...
+ * @param {boolean} open State of the <Admin/> Sidebar
+ * @param {string} title The title of the AppBar
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/AppBar.js]: no, it's not used - you can remove it.

File: packages/ra-ui-materialui/src/layout/AppBar.js [hunk 6]
```diff
@@ -47,12 +47,39 @@ const useStyles = makeStyles(
...
+ * @param {Function} userMenu A custom UserMenu component for the AppBar. UserMenu component by default
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/AppBar.js]: Not Function, but Element

File: packages/ra-ui-materialui/src/layout/AppBar.js (hunk 3 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/layout/AppBar.js]: Ok. I've copied it from another ra components actually.

**Comment by WiXSL** [comment]
Toolbar `disableGutters` prop isn't set anywhere.
Should I remove it, or take it from the AppBar props?

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**